### PR TITLE
fix: make _convert_to_structure_to_list return a type instead of an ElementaryType's `type` field

### PR DESCRIPTION
**Pull request by kevinclancy**
Fix https://github.com/crytic/slither/issues/1918

### CODE REVIEW

**Comment by 0xalpharush** [comment]
Would you mind making this against the dev branch please?

File: slither/slithir/convert.py [hunk 1]
```diff
@@ -1583,7 +1583,10 @@ def _convert_to_structure_to_list(return_type: Type) -> List[Type]:
     # }
     if isinstance(return_type, (MappingType, ArrayType)):
         return []
-    return [return_type.type]
+    if isinstance(return_type, (ElementaryType, UserDefinedType, TypeAlias)):
+        return [return_type]
+
+    assert False
```
Comment (0xalpharush) [inline slither/slithir/convert.py:1589]: Could we change the isinstance check to an assert instead of asserting False?

Comment (0xalpharush) [inline slither/slithir/convert.py:1589]:
```suggestion
    assert isinstance(return_type, (ElementaryType, UserDefinedType, TypeAlias)):
    return [return_type]
```

**Comment by kevinclancy** [comment]
Yes. Done.

**Pull request merged**